### ISSUE TITLE
Defining Inputs and Outputs

### ISSUE THREAD

**Issue by Garygunn94** [ibab/tensorflow-wavenet#216]
Hi there. Firstly I'm relatively new to neural networks so apologies for that. I'm doing some research into audio source separation of raw audio waveforms. My data consists of input mixed signals and output single source signals with the idea being that the model is trained to separate a consistant audio source from mixed signals. Wavenet seems to be a good fit for investigating this, however im struggling to figure out how to set hte input and output of the network. I understand that the idea of wavenet is as a generation network, but I was wondering if/how to train the network with defined input and output.

Any help would be appreciated!

**Comment by akademi4eg** [comment]
Wavenet is an autoregressive model, it tries to continue given sequence. So, in current revision you can only feed inputs for the network and it will figure out outputs itself by shifting input by 1 sample. To do source separation you could try to modify loss calculation so that at train time it accepts mixed signal as input and substitute shifting with target signal. It would take some time coding, yet it might work.

Feel free to submit a pull request if you try this. :)

**Comment by Garygunn94** [comment]
Great thanks for your help, I'll look into it and see what I can do!
